test/e2e/elasticsearch_test.go

```diff
@@ -113,104 +134,192 @@ func (suite *ElasticSearchTestSuite) TestSimpleProd() {
+		jaegerInstance.Spec.Collector.Image = otelCollectorImage
+		jaegerInstance.Spec.Collector.Config = v1.NewFreeForm(getOtelConfigForHealthCheckPort("14269"))
+	}
 
-		err := fw.Client.Create(context.Background(), jaegerInstance, &framework.CleanupOptions{TestContext: ctx, Timeout: timeout, RetryInterval: retryInterval})
-		require.NoError(t, err, "Error deploying Jaeger")
-		defer undeployJaegerInstance(jaegerInstance)
+	// deploy jaeger services
+	logrus.Infof("Creating jaeger services for es index cleaner test: %s", jaegerInstanceName)
+	createESSelfProvDeployment(jaegerInstance, jaegerInstanceName, namespace)
+	defer undeployJaegerInstance(jaegerInstance)
+
+	// generate spans and service for last 45 days
+	currentDate := time.Now()
+	indexDateLayout := "2006-01-02"
+	// enable port forward for collector port
+	logrus.Info("Enabling collector port forward")
+	fwdPortColl, closeChanColl := CreatePortForward(namespace, jaegerInstanceName+"-collector", "collector", []string{fmt.Sprintf(":%d", jaegerCollectorPort)}, fw.KubeConfig)
+	defer fwdPortColl.Close()
+	defer close(closeChanColl)
+	// get localhost collector port
+	colPorts, err := fwdPortColl.GetPorts()
+	require.NoError(t, err)
+	localPortColl := colPorts[0].Local
+
+	logrus.Infof("Generating spans and services for the last %d days", indexHistoryDays)
+	for day := 0; day < indexHistoryDays; day++ {
+		spanDate := currentDate.AddDate(0, 0, -1*day)
+		stringDate := spanDate.Format(indexDateLayout)
+		// get tracing client
+		serviceName := fmt.Sprintf("%s_%s", jaegerInstanceName, stringDate)
+		tracer, closer, err := getTracerClientWithCollectorEndpoint(serviceName, fmt.Sprintf("http://localhost:%d/api/traces", localPortColl))
+		require.NoError(t, err)
+		// generate span
+		tracer.StartSpan("span-index-cleaner", opentracing.StartTime(spanDate)).
+			SetTag("jaeger-instance", jaegerInstanceName).
+			SetTag("test-case", t.Name()).
+			SetTag("string-date", stringDate).
+			FinishWithOptions(opentracing.FinishOptions{FinishTime: spanDate.Add(time.Second)})
+		closer.Close()
+	}
 
-		err = e2eutil.WaitForDeployment(t, fw.KubeClient, namespace, jaegerInstanceName, 1, retryInterval, timeout)
-		require.NoError(t, err, "Error waiting for deployment")
+	// esIndexData struct is used to keep index data in simple format
+	// will be useful for the validations
+	type esIndexData struct {
+		IndexName string    // original index name
+		Type      string    // index type. span or service?
+		Prefix    string    // prefix of the index
+		Date      time.Time // index day/date
+	}
 
-		// create span, then make sure indices have been created
-		AllInOneSmokeTest(jaegerInstanceName)
+	// function to get indices
+	// returns in order: serviceIndices, spansIndices
+	getIndices := func() ([]esIndexData, []esIndexData) {
+		// get indices from es node
+		esIndices, err := getEsIndices()
+		require.NoError(t, err)
+		fmt.Println("indices found on rest api response:", len(esIndices))
```

Comment:
It would be better to be consistent and use a logrus.Infof here instead of fmt.Println